rocAL/rocAL_pybind/example/new_api/rocAL_api_python_unittest.py

```diff
@@ -0,0 +1,183 @@
+from __future__ import division
+from __future__ import print_function
+import random
+from amd.rocal.plugin.pytorch import RALIClassificationIterator
+
+from amd.rocal.pipeline import Pipeline
+import amd.rocal.fn as fn
+import amd.rocal.types as types
+import sys
+
+def draw_patches(img,idx):
+    #image is expected as a tensor, bboxes as numpy
+    import cv2
+    image = img.detach().numpy()
+    image = image.transpose([1,2,0])
+    image = cv2.cvtColor(image, cv2.COLOR_RGB2BGR)
+    cv2.imwrite(str(idx)+"_"+"train"+".png", image)
+
+
+def main():
+    if  len(sys.argv) < 5:
+        print ('Please pass image_folder augmentation_number cpu/gpu batch_size')
+        exit(0)
+    data_path = sys.argv[1]
+    augmentation_num = int(sys.argv[2])
+
+    if(sys.argv[3] == "cpu"):
+        _rali_cpu = True
+    else:
+        _rali_cpu = False
+
+    bs = int(sys.argv[4])
+    nt = 1
+    di = 0
+    random_seed = random.SystemRandom().randint(0, 2**32 - 1)
+    crop=300
+
+    pipe = Pipeline(batch_size=bs, num_threads=nt,device_id=di, seed=random_seed, rali_cpu=_rali_cpu)
+    output_set=0
+    local_rank = 0
+    world_size = 1
+
+
+    rali_cpu= True
+    rali_device = 'cpu' if rali_cpu else 'gpu'
+    decoder_device = 'cpu' if rali_cpu else 'mixed'
+
+    crop_size = 300
+
+    with pipe:
+        jpegs, labels = fn.readers.file(file_root=data_path, shard_id=local_rank, num_shards=world_size, random_shuffle=True)
+        images = fn.decoders.image(jpegs, file_root=data_path, device=decoder_device, output_type=types.RGB, shard_id=0, num_shards=1, random_shuffle=True)
+        # images = fn.decoders.image_random_crop(jpegs, device=decoder_device, output_type=types.RGB,
+        #                                             device_memory_padding=device_memory_padding,
+        #                                             host_memory_padding=host_memory_padding,
+        #                                             random_aspect_ratio=[0.8, 1.25],
+        #                                             random_area=[0.1, 1.0],
+        #                                             num_attempts=100,
+        #                                             file_root=data_path, shard_id=local_rank, num_shards=world_size, random_shuffle=True)
+        images = fn.resize(images, device=rali_device, resize_x=crop, resize_y=crop)
+        # flip_coin = fn.random.coin_flip(probability=0.5)
+        # images = fn.crop_mirror_normalize(images, device="gpu",
+        #                                 output_dtype=types.FLOAT,
+        #                                 output_layout=types.NCHW,
+        #                                 crop=(crop, crop),
+        #                                 mirror=flip_coin,
+        #                                 image_type=types.RGB,
+        #                                 mean=[0,0,0],
+        #                                 std=[1,1,1])
+
+        if augmentation_num == 0 or augmentation_num == 1:
```

Comment:
please change augmentation_num to string (name so it is more clear)

Reply:
I was working on modifying the test suite for python. Added the above change in latest commit